### PR TITLE
Initial commit for data source automation

### PR DESCRIPTION
**Pull request by chelnak**

### CODE REVIEW

File: datasources/New-SqlDatabaseServiceAccount.ps1 [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                $Role = @"
+                ALTER ROLE db_datawriter
+                ADD MEMBER "$($SqlServiceAccountName)"
+"@
+                break
+
+            }
+
+            'RWE' {
+                $Role = @"
```
Comment (jack-education) [inline datasources/New-SqlDatabaseServiceAccount.ps1]:
```suggestion
                $Role = @"
                ALTER ROLE db_datareader
                ADD MEMBER "$($SqlServiceAccountName)"
```

File: datasources/New-SqlDatabaseServiceAccount.ps1 [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            $ServiceAccountPassword = Get-RandomPassword
+            $SecureAccountPassword = $ServiceAccountPassword | ConvertTo-SecureString -AsPlainText -Force
+            $null = Set-AzKeyVaultSecret -VaultName $KeyVaultName -Name $ServiceAccountSecretName -SecretValue $SecureAccountPassword -Verbose:$VerbosePreference
+        }
+
+        switch($SqlServiceAccountRole) {
+            'R' {
+                $Role = @"
+                ALTER ROLE db_datareader
+                ADD MEMBER "$($SqlServiceAccountName)"
```
Comment (jack-education) [inline datasources/New-SqlDatabaseServiceAccount.ps1]:
```suggestion
                ADD MEMBER "$($SqlServiceAccountName)"
                IF IS_ROLEMEMBER('db_datawriter','$($SqlServiceAccountName)') = 1
                ALTER ROLE db_datawriter
                DROP MEMBER "$($SqlServiceAccountName)"
```

File: datasources/helpers/New-SqlDatabaseServiceAccount.ps1 [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            $ServiceAccountPassword = Get-RandomPassword
+            $SecureAccountPassword = $ServiceAccountPassword | ConvertTo-SecureString -AsPlainText -Force
+            $null = Set-AzKeyVaultSecret -VaultName $KeyVaultName -Name $ServiceAccountSecretName -SecretValue $SecureAccountPassword -Verbose:$VerbosePreference
+        }
+
+        switch ($SqlServiceAccountRole) {
+            'R' {
+                $Role = @"
+                ALTER ROLE db_datareader
+                ADD MEMBER "$($SqlServiceAccountName)"
```
Comment (jack-education) [inline datasources/helpers/New-SqlDatabaseServiceAccount.ps1]:
```suggestion
                ADD MEMBER "$($SqlServiceAccountName)"
                ALTER ROLE db_datawriter
                DROP MEMBER "$($SqlServiceAccountName)"
                REVOKE EXECUTE FROM "$($SqlServiceAccountName)"
```

File: datasources/helpers/New-SqlDatabaseServiceAccount.ps1 [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+"@
+                break
+            }
+
+            'RW' {
+                $Role = @"
+                ALTER ROLE db_datawriter
+                ADD MEMBER "$($SqlServiceAccountName)"
```
Comment (jack-education) [inline datasources/helpers/New-SqlDatabaseServiceAccount.ps1]:
```suggestion
                ADD MEMBER "$($SqlServiceAccountName)"
                REVOKE EXECUTE FROM "$($SqlServiceAccountName)"
```

File: datasources/helpers/New-SqlDatabaseServiceAccount.ps1 [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+"@
+                break
+            }
+
+            'RWE' {
+                $Role = @"
```
Comment (jack-education) [inline datasources/helpers/New-SqlDatabaseServiceAccount.ps1]:
```suggestion
                $Role = @"
                ALTER ROLE db_datareader
                ADD MEMBER "$($SqlServiceAccountName)"
```

File: datasources/helpers/Invoke-GrafanaDataSourceRestMethod.ps1 [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+        $Headers = @{
+            "Accept"        = "application/json"
+            "Content-Type"  = "application/json"
+            "Authorization" = "Bearer $GrafanaApiToken"
+        }
+
+        $DataSourceName = ($Payload | ConvertFrom-Json).Name
+
+        if (!$GrafanaBaseUri.EndsWith("/")) {
```
Comment (jack-education) [inline datasources/helpers/Invoke-GrafanaDataSourceRestMethod.ps1]:
```suggestion
        if ($GrafanaBaseUri.EndsWith("/")) {
```

**Comment by chelnak** [comment]
New-SqlDatabaseServiceAccount will eventually be consumed from platform-automation.

File: datasources/helpers/New-SqlDatabaseServiceAccount.ps1 [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+            'R' {
+                $Role = @"
+                ALTER ROLE db_datareader
+                ADD MEMBER "$($SqlServiceAccountName)"
+"@
+                break
+            }
+
+            'RW' {
+                $Role = @"
```
Comment (jack-education) [inline datasources/helpers/New-SqlDatabaseServiceAccount.ps1]:
```suggestion
                $Role = @"
                ALTER ROLE db_datareader
                ADD MEMBER "$($SqlServiceAccountName)"
```

File: datasources/helpers/New-SqlDatabaseServiceAccount.ps1 [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+            $ServiceAccountPassword = Get-RandomPassword
+            $SecureAccountPassword = $ServiceAccountPassword | ConvertTo-SecureString -AsPlainText -Force
+            $null = Set-AzKeyVaultSecret -VaultName $KeyVaultName -Name $ServiceAccountSecretName -SecretValue $SecureAccountPassword -Verbose:$VerbosePreference
+        }
+
+        switch ($SqlServiceAccountRole) {
+            'R' {
+                $Role = @"
+                ALTER ROLE db_datareader
+                ADD MEMBER "$($SqlServiceAccountName)"
```
Comment (jack-education) [inline datasources/helpers/New-SqlDatabaseServiceAccount.ps1]:
```suggestion
                ADD MEMBER "$($SqlServiceAccountName)"
                ALTER ROLE db_datawriter
                DROP MEMBER "$($SqlServiceAccountName)"
                REVOKE EXECUTE FROM "$($SqlServiceAccountName)"
```

**Pull request merged**